### ISSUE TITLE
Agents ignoring offline messages

### ISSUE THREAD

**Issue by dmvieira** [javipalanca/spade#84]
* SPADE version: 3.1.4
* Python version: 3.7.7
* Operating System: Ubuntu 16.04

### Description

When you send messages and agents are offline or in unavailable state are ignored when agent back to available state.

### What I Did

- Send message and start receiver agent
- ```set_unavailable()``` agent, send a message to this agent, than ```set_available()``` agent

**Comment by javipalanca** [comment]
Yes, this is a feature of XMPP. You cannot send a message to an offline agent unless your server supports and has active the XEP-0136: Message Archiving.

**Comment by dmvieira** [comment]
But I'm using Ejabberd and its showing offiline messages, but when agent is up all messages are consumed and ignored

**Comment by javipalanca** [comment]
Hi. I was guessing why this issue happens and it is because supporting offline messages (XEP-0013: Flexible Offline Message Retrieval) requires a proactive action from the client (and also an implementation of the extension in the server side).
This action is not still included in SPADE, but since it is a good feature I'll try to include it in future releases following https://xmpp.org/extensions/xep-0013.html.
